App Submission: ioBroker

amariwan opened the pull request:
# App Submission
ioBroker

### SVG icon
![logo](https://github.com/user-attachments/assets/1608daa8-5981-4d26-87ca-bab26437f1e3)


### Gallery images
![Dashboard](https://www.iobroker.net/de/admin/media/ADMIN_Uebersicht.png)
![Adapters](https://www.iobroker.net/de/admin/media/ADMIN_Adapter_Kachel_versions.png)
![VIS](https://www.iobroker.net/assets/screen11-DGRrXChA.jpg)

### I have tested my app on:
- [] umbrelOS on a Raspberry Pi
- [] umbrelOS on an Umbrel Home
- [X] umbrelOS on Linux VM

amariwan committed:
Add initial configuration files for ioBroker setup

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update docker-compose.yml

amariwan committed:
Update docker-compose.yml

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update docker-compose.yml

amariwan committed:
Delete iobroker/hooks/pre-start

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update umbrel-app.yml

amariwan committed:
Update docker-compose.yml

amariwan committed:
Update umbrel-app.yml

al-lac commented on iobroker/docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+  app_proxy:
+    environment:
+      APP_HOST: iobroker_server_1
+      APP_PORT: 8097
```
The port where the server is reachable has to be specified here.
```suggestion
      APP_PORT: 8081
```

amariwan replied:
I actually can’t use port 8098 here, it’s already taken by another Umbrel app.

al-lac commented on iobroker/docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+    volumes:
+      - ${APP_DATA_DIR}/data:/opt/iobroker
+      - ${UMBREL_ROOT}/data/storage/downloads:/media
+      - /dev:/dev
```
This should not be needed to mount adapters.
```suggestion
```

al-lac commented on iobroker/umbrel-app.yml:
```diff
@@ -0,0 +1,49 @@
+version: '2025.6.1'
+tagline: Open source home automation platform for every use case
+description: >-
+     🧠 **ioBroker** is the Swiss Army knife of home automation.
+
+     With support for **hundreds of devices and protocols**, including Zigbee, Z-Wave, KNX, MQTT, EnOcean, Shelly, Philips Hue, Sonos, Alexa, and more – ioBroker gives you full control over your smart home from a clean, browser-based dashboard.
+
+     💡 Want automation? Want custom dashboards? Want privacy-first logic without cloud lock-in? You've found your home base. ioBroker runs **entirely local** on your Umbrel node. Your data stays yours — 100%.
+
+     --- 🔧 **Features at a glance:**
+
+     - 🌐 Sleek web UI to configure devices and automations
+     - 🧩 400+ adapters and integrations from the ioBroker community
+     - 🛜 Local-first architecture (no vendor cloud required)
+     - 🔐 Full control of your data, locally encrypted
+     - 🧱 Visual logic editor (Blockly/JavaScript-based flows)
+
+     --- 🚀 **Setup instructions:**
+
+     - Open the dashboard at port `8097`
+     - On first launch, follow the **setup wizard**
+     - USB devices (Zigbee, etc.) may require manual mapping in Docker
+     - Use the Adapter tab to explore and install integrations
+
+     For power users: install VIS, Node-RED, BackItUp, Alexa support and more — all through the integrated adapter system.
+
+     🛠️ Learn more at [iobroker.net](https://www.iobroker.net) or join the community: [forum.iobroker.net](https://forum.iobroker.net)
+
```
Adding proper spacing and formatting.
```suggestion
     🧠 **ioBroker** is the Swiss Army knife of home automation.


     With support for **hundreds of devices and protocols**, including Zigbee, Z-Wave, KNX, MQTT, EnOcean, Shelly, Philips Hue, Sonos, Alexa, and more – ioBroker gives you full control over your smart home from a clean, browser-based dashboard.


     💡 Want automation? Want custom dashboards? Want privacy-first logic without cloud lock-in? You've found your home base. ioBroker runs **entirely local** on your Umbrel node. Your data stays yours — 100%.

     - 🔧 **Features at a glance**
       - 🌐 Sleek web UI to configure devices and automations
       - 🧩 400+ adapters and integrations from the ioBroker community
       - 🛜 Local-first architecture (no vendor cloud required)
       - 🔐 Full control of your data, locally encrypted
       - 🧱 Visual logic editor (Blockly/JavaScript-based flows)

     - 🚀 **Setup instructions**
       - Open the dashboard at port `8098`
       - On first launch, follow the **setup wizard**
       - USB devices (Zigbee, etc.) may require manual mapping in Docker
       - Use the Adapter tab to explore and install integrations

     For power users: install VIS, Node-RED, BackItUp, Alexa support and more — all through the integrated adapter system.


     🛠️ Learn more at [iobroker.net](https://www.iobroker.net) or join the community: [forum.iobroker.net](https://forum.iobroker.net)
```

al-lac commented on iobroker/umbrel-app.yml:
```diff
@@ -0,0 +1,49 @@
+dependencies: []
+repo: https://github.com/ioBroker/ioBroker
+support: https://forum.iobroker.net
+port: 8097
```
Use a different port as 8097 is already used in `blockstream-blind-oracle` app
```suggestion
port: 8098
```

al-lac reviewed (changes requested):
Hey @amariwan,

thanks for the great submission! 🧰 

I only got the app running after removing the network_mode. Also suggested some adaptions to get the app ready. Please take a look and test if the app is working for you then.

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

al-lac reviewed (changes requested):
Hey @amariwan,

thanks for making the adaptions. And sorry, missed the fact that port `8098` is already in use. Added some more comments.

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

al-lac reviewed (changes requested):
Hey @amariwan,

thanks for adapting the changes.

I found some more things that could be improved. Please take a look and get back to me!

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

amariwan committed:
Update iobroker/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

al-lac commented:
Any update @amariwan? 🖼️

al-lac commented:
Checking in again @amariwan, do you have any update on this app submission?

al-lac commented:
Hey @amariwan, closing this for now. Feel free to re-open if you want to work on it again.

amariwan commented:
Hey @al-lac,

sorry for the long silence. I'd like to continue working on the ioBroker submission and address the remaining feedback.

I currently don't see an option to reopen the PR myself. Could you please reopen it for me?

Thanks!
Aland

al-lac commented:
Hey @amariwan,

No worries!

Reopened it, @nmfretz and @kaichappel will help you to get this over the finish line.

amariwan commented:
Hey @nmfretz @kaichappel — the ioBroker submission is back in shape and I’ve reworked it against the current Umbrel packaging guidelines.

Main updates:

* `iobroker/iobroker:v11.1.0` pinned to a verified multi-arch digest (`amd64` + `arm64`)
* Umbrel host port `8104` -> `app_proxy` -> internal ioBroker Admin port `8081`
* persistent state kept under `${APP_DATA_DIR}/data:/opt/iobroker`
* removed unnecessary privileged/host/device/D-Bus access
* manifest and App Store copy updated
* current Umbrel linter and image checks pass

I’d appreciate another review when you have time. If you spot anything else, I’m happy to fix it.

amariwan commented on iobroker/docker-compose.yml:
```diff
@@ -0,0 +1,16 @@
+      APP_PORT: 8081
+
+  server:
+    image: buanet/iobroker:v9.1.2@sha256:84f0699976f54fde3a9e321240f94d7d526ba9cc73e3f2f669a8facc6f4da294
```
Hey @al-lac — resolved: package now uses canonical image `iobroker/iobroker:v11.1.0@sha256:840ee2408f2d2474482cbc836dec6b349c011f8c75b366cb01164b95d7d37ab9` with immutable **multi-arch manifest-list** digest (verified via `docker buildx imagetools inspect`: `linux/amd64 sha256:b1b0fca5...` + `linux/arm64 sha256:f1424773...`). Previous `buanet/iobroker:latest` removed. Version `11.1.0` is current stable (2026-08-26, matches `latest`/`latest-v11`). Future maintainers: check https://hub.docker.com/r/iobroker/iobroker/tags for latest `vX.Y.Z`, release notes at https://github.com/buanet/ioBroker.docker/releases and https://docs.buanet.de/iobroker-docker-image/docs/, then resolve digest with `docker buildx imagetools inspect iobroker/iobroker:<tag>`.

amariwan commented on iobroker/umbrel-app.yml:
```diff
@@ -0,0 +1,49 @@
+id: iobroker
+category: automation
+name: ioBroker
+version: '2025.6.1'
```
Hey @al-lac — versioning clarified: `umbrel-app.yml:version` is the Docker image tag `11.1.0` (image bundles js-controller + admin; not the controller version alone). Verified current stable is **v11.1.0** (2026-08-26) via Hub tags. Pinned digest is the multi-arch index `sha256:840ee240...` (`amd64+arm64` verified). For future updates see https://hub.docker.com/r/iobroker/iobroker/tags (latest `vX.Y.Z`), changelog at https://github.com/buanet/ioBroker.docker/releases / https://docs.buanet.de/iobroker-docker-image/docs/, and digest via `docker buildx imagetools inspect iobroker/iobroker:<tag>`.
